src/toil/test/docs/scripts/example_cachingbenchmark.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+# example_cachingbenchmark.py: Empirically inspect Toil caching
+
+"""
+This workflow collects statistics about caching.
+
+Invoke like:
+
+    python examples/example_cachingbenchmark.py ./jobstore --realTimeLogging --logInfo --disableCaching False
+    
+    python examples/example_cachingbenchmark.py ./jobstore --realTimeLogging --logInfo --disableCaching True
+    
+    python examples/example_cachingbenchmark.py aws:us-west-2:cachingjobstore --realTimeLogging --logInfo --disableCaching False
+    
+    python examples/example_cachingbenchmark.py aws:us-west-2:cachingjobstore --realTimeLogging --logInfo --disableCaching True
+        
+"""
+
+import argparse
+import sys
+import os
+import socket
+import time
+import random
+import collections
+
+from toil.common import Toil
+from toil.job import Job
+from toil.leader import FailedJobsException
+from toil.realtimeLogger import RealtimeLogger
+
+def main():
+    parser = argparse.ArgumentParser(description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
+    
+    parser.add_argument('--minSleep', type=int, default=1,
+        help="Minimum seconds to sleep")
+    
+    Job.Runner.addToilOptions(parser)
+    
+    options = parser.parse_args(sys.argv[1:])
+
+    root_job = Job.wrapJobFn(root, options)
+
+    with Toil(options) as toil:
+        results = toil.start(root_job)
+        
+    print("Caching results:")
+    print(results)
+
+
+def root(job, options):
+    # Make a file
+    with job.fileStore.writeGlobalFileStream() as (stream, file_id):
+        stream.write(('This is a test of the Toil file caching system. ' +
+            'Had this been an actual file, its contents would have been more interesting.').encode('utf-8'))
+    
+    child_rvs = []
+    for i in range(100):
```

Comment:
How long does this test take to run?

Reply:
I timed it at about 5 minutes on our cluster.